### PR TITLE
Recipe backend

### PR DESCRIPTION
**Pull request by Lexseal**
### Features Added/Issues Addressed
1. Reformatted code so linter doesn't scream and CI isn't broken
2. Addressed #56 by added `recipe_data` as our recipe data serialization template
3. Fixed intolerances bug and improved `get_recipe()` function
4. Added documentation
5. Added tests

### Brief Description of Solution
Fixed linter by adding a new rule that sends unused variables to warnings instead of errors. Additionally, ran `npm run lint` to fix spacing and semicolon issues.

Improved `get_recipe()` by
* separating out the fetch raw json and filtering into different functions
* combined intolerance search with regular search
* returns empty lists instead of undefined when recipe not found
* the recipe returned will follow the `recipe_data` object. So in the future we can easily add more fields by changing `recipe_data`

### Checklist
- [x] Added new tests (You know the code best to make new tests)
- [x] Resolved merge conflicts (Prefer merge over rebase)
- [x] Status check passed (Linted and tested)
- [ ] Reviewed (At least one reviewer's approval)

### Caveat/Special Instructions for Team
@prorick @sora1998 @meshachadoe @yiw008 and anyone else working on recipe cards,
please take a look at [this file](https://github.com/cse110-fa21-group30/cse110-fa21-group30/blob/aed2f788df799e4c8df5337616bbb978260d88b2/source/assets/scripts/recipe-data.js).
Feel free to comment below if that is a good enough format for what we do

I also upgraded our spoonacular subscription to the basic tier (150/day + $0.007 per request over 150). If we use considerably more (300+ requests per day), please let me know and I can upgrade it again : )

### CODE REVIEW

**Comment by Lexseal** [comment]
Also I noticed that most of the PRs are approved and merged without people commenting on code at all. Additionally, most status checks are ignored. Please go to the [files changed tab](https://github.com/cse110-fa21-group30/cse110-fa21-group30/pull/73/files) to review and comment in the future!

File: .eslintrc.json [hunk 1]
```diff
@@ -24,6 +24,10 @@
         "semi": [
             "error",
             "always"
+        ],
+        "no-unused-vars": [
+            "warn",
```
Comment (Lexseal) [inline .eslintrc.json:29]: now unused stuff doesn't break CI

File: package.json [hunk 2]
```diff
@@ -23,6 +23,7 @@
     "eslint": "^8.1.0",
     "jest": "^27.3.1",
     "jest-puppeteer": "^6.0.0",
+    "node-fetch": "^3.1.0",
```
Comment (Lexseal) [inline package.json:26]: For Jest to run in node

**Comment by Lexseal** [comment]
@Zijie-Lei Can you take a look at the codacy stuff? Thank you!

**Comment by Xubangbangg** [comment]
So now we can get the recipe json file and we still need to parse it and handle the data for our own recipe card right?

**Comment by Lexseal** [comment]
> So now we can get the recipe json files. We still need to parse it and handle the data for our own recipe card right?

If you are just using the data, you shouldn't need to parse it because that has been done. For example, if you just want to display a recipe card, then you can just grab information from the object.

On the other hand, if you want to save user recipes, you should store the user input in the format specified by the `recipe_data` object.

Let me know if you have more questions : )

**Review by Xubangbangg — APPROVED** [review]
looks good

**Pull request merged**